### PR TITLE
i3pystatus: 2019-06-10 -> 2020-07-14; fix desktop notifications

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/i3/pystatus.nix [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
+  ]; # libpulseaudio.so is loaded manually
+
+  postConfigure = ''
+    makeWrapperArgs+=("--set" "GI_TYPELIB_PATH" "$GI_TYPELIB_PATH")
+  '';
+
+  buildInputs = [ libpulseaudio libnotify gobject-introspection ];
 
   postInstall = ''
     makeWrapper ${python3Packages.python.interpreter} $out/bin/${pname}-python-interpreter \
```
Comment (igsha) [inline pkgs/applications/window-managers/i3/pystatus.nix:41]: This is not needed any more https://github.com/enkore/i3pystatus/pull/400. Please, remove `postInstall` section

File: pkgs/applications/window-managers/i3/pystatus.nix [hunk 2]
```diff
@@ -1,30 +1,46 @@
...
+, extraLibs ? [] }:
 
-python3Packages.buildPythonApplication rec {
+let
+  ldWrapperSuffix = "--suffix LD_LIBRARY_PATH : \"${stdenv.lib.makeLibraryPath [ libpulseaudio ]}\"";
+in python3Packages.buildPythonApplication rec {
   # i3pystatus moved to rolling release:
   # https://github.com/enkore/i3pystatus/issues/584
-  version = "unstable-2019-06-10";
+  version = "unstable-2020-07-14";
```
Comment (igsha) [inline pkgs/applications/window-managers/i3/pystatus.nix:14]: Are you sure about the date? May be 2020-06-12 (the date of the commit)?

File: pkgs/applications/window-managers/i3/pystatus.nix [hunk 3]
```diff
@@ -1,30 +1,46 @@
...
 
-  src = fetchFromGitHub
-  {
+  src = fetchFromGitHub {
     owner = "enkore";
     repo = "i3pystatus";
-    rev = "56ce08d0ff8d5d64950d6b588ebede35a95e0ce2";
-    sha256 = "12938860jbcly1xwhd71jvy2dff28pwv9kqh6mab1859148bzmcg";
+    rev = "dad5eb0c5c8a2ecd20c37ade4732586c6e53f44b";
+    sha256 = "12938860jbcly1xwhd71jvy2dff28pwv9kqh6mab1859148bzmc0";
```
Comment (igsha) [inline pkgs/applications/window-managers/i3/pystatus.nix:21]: nix-prefetch-url shows different sha:
```
nix-prefetch-url --unpack https://github.com/enkore/i3pystatus/archive/dad5eb0c5c8a2ecd20c37ade4732586c6e53f44b.tar.gz
unpacking...
[0.2 MiB DL]
path is '/nix/store/hs5zg81dais8w6vm6ingb42bf7abszh1-dad5eb0c5c8a2ecd20c37ade4732586c6e53f44b.tar.gz'
18ygvkl92yr69kxsym57k1mc90asdxpz4b943i61qr0s4fc5n4mq
```

File: pkgs/applications/window-managers/i3/pystatus.nix [hunk 4]
```diff
@@ -1,30 +1,46 @@
...
-  ldWrapperSuffix = "--suffix LD_LIBRARY_PATH : \"${libpulseaudioPath}\"";
   # LC_TIME != C results in locale.Error: unsupported locale setting
-  makeWrapperArgs = [ "--set LC_TIME C" ldWrapperSuffix ]; # libpulseaudio.so is loaded manually
+  makeWrapperArgs = [
+    "--set" "LC_TIME" "C"
+    ldWrapperSuffix
+  ]; # libpulseaudio.so is loaded manually
+
+  postConfigure = ''
+    makeWrapperArgs+=("--set" "GI_TYPELIB_PATH" "$GI_TYPELIB_PATH")
```
Comment (igsha) [inline pkgs/applications/window-managers/i3/pystatus.nix:35]: According to [documentation](https://nixos.org/nixpkgs/manual/#ssec-gnome-hooks) you just need to add `wrapGAppsHook` into `buildInputs`.

**Review by igsha — CHANGES_REQUESTED** [review]

File: pkgs/applications/window-managers/i3/pystatus.nix (hunk 4 above)
Comment (Mic92) [reply pkgs/applications/window-managers/i3/pystatus.nix]: It did not really pick `GI_TYPELIB_PATH` for me. I assume I would need to add more dependencies like gtk3 to `buildInputs` but i3pystatus does not really need those.

File: pkgs/applications/window-managers/i3/pystatus.nix (hunk 3 above)
Comment (Mic92) [reply pkgs/applications/window-managers/i3/pystatus.nix]: fixed.

File: pkgs/applications/window-managers/i3/pystatus.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/applications/window-managers/i3/pystatus.nix:40]: I tried that however `i3pystatus` does not allow to import modules from the same directory i.e.: https://github.com/Mic92/dotfiles/blob/master/home/.i3/i3status_bottom.py#L7 and https://github.com/Mic92/dotfiles/blob/master/home/.i3/color.py

File: pkgs/applications/window-managers/i3/pystatus.nix [hunk 5]
```diff
@@ -1,31 +1,45 @@
...
     repo = "i3pystatus";
-    rev = "56ce08d0ff8d5d64950d6b588ebede35a95e0ce2";
-    sha256 = "12938860jbcly1xwhd71jvy2dff28pwv9kqh6mab1859148bzmcg";
+    rev = "dad5eb0c5c8a2ecd20c37ade4732586c6e53f44b";
+    sha256 = "18ygvkl92yr69kxsym57k1mc90asdxpz4b943i61qr0s4fc5n4mq";
   };
 
-  propagatedBuildInputs = with python3Packages; [ keyring colour netifaces psutil basiciw ] ++
-    [ libpulseaudio ] ++ extraLibs;
+  buildInputs = [ libpulseaudio libnotify gobject-introspection ];
```
Comment (igsha) [inline pkgs/applications/window-managers/i3/pystatus.nix:23]: `libpulseaudio` goes to `LD_LIBRARY_PATH`, `gobject-introspection` to `GI_TYPELIB_PATH`. But what about `libnotify`? Are you sure that it is not needed for propagation at runtime?

File: pkgs/applications/window-managers/i3/pystatus.nix (hunk 1 above)
Comment (igsha) [reply pkgs/applications/window-managers/i3/pystatus.nix:40]: ок

File: pkgs/applications/window-managers/i3/pystatus.nix (hunk 5 above)
Comment (Mic92) [reply pkgs/applications/window-managers/i3/pystatus.nix:23]: `libnotify` is also in in `GI_TYPELIB_PATH`:
```
$ grep libnotify /nix/store/0cvngq5xlihgmal336xs784pmr6sdirx-i3pystatus-unstable-2020-06-12/bin/i3pystatus-python-interpreter
export GI_TYPELIB_PATH='/nix/store/lwzdlszgqahazib2i1zjkqdf8wfjx820-gdk-pixbuf-2.40.0/lib/girepository-1.0:/nix/store/gsjhvi2nbpbfkf6dcz6ibkl4pn3ksym5-libnotify-0.7.9/lib/girepository-1.0:/nix/store/d6sbxgycqpabzf87pymvk2s0d4dlzbqn-gobject-introspection-1.64.1/lib/girepository-1.0'
```

**Pull request merged**